numpy/_core/tests/test_numerictypes.py

```diff
@@ -616,6 +616,35 @@ def test_names_are_undersood_by_dtype(self, t):
         assert np.dtype(t.__name__).type is t
 
 
+class TestScalarTypeOrder:
+    @pytest.mark.parametrize(('a', 'b'), [
+        # signedinteger
+        (np.byte, np.short),
+        (np.short, np.intc),
+        (np.intc, np.long),
+        (np.long, np.longlong),
+        # unsignedinteger
+        (np.ubyte, np.ushort),
+        (np.ushort, np.uintc),
+        (np.uintc, np.ulong),
+        (np.ulong, np.ulonglong),
+        # floating
+        (np.half, np.single),
+        (np.single, np.double),
+        (np.double, np.longdouble),
+        # complexfloating
+        (np.csingle, np.cdouble),
+        (np.cdouble, np.clongdouble),
+        # flexible
+        (np.bytes_, np.str_),
+        (np.str_, np.void),
+        # bouncy castles
+        (np.datetime64, np.timedelta64),
```

Comment:
Total quibble but I think you can write this with just a single static list and then generate these tuples with e.g. `itertools.pairwise`.

Reply:
I wanted to do that at first. But I then decided against that because I didn't want to set the (IMO arbitrary) "kind" group ordering in stone (bool < complexfloating < np.floating < np.signedinteger < bouncy castles < object < flexible < unsignedinteger).